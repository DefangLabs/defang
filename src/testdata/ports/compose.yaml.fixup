{
  "long": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "bogus",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 82,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  "long-published": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "bogus",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 83,
        "published": "8083",
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  "short": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "bogus",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 80,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  "short-published": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "bogus",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 81,
        "published": "8081",
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  "short-udp": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "bogus",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 84,
        "protocol": "udp"
      }
    ]
  },
  "short-udp-published": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "bogus",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 85,
        "published": "8085",
        "protocol": "udp"
      }
    ]
  }
}