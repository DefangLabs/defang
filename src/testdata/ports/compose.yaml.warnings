 ! port 81: 'published' should be equal to 'target'; ignoring 'published: 8081'
 ! port 83: 'published' should be equal to 'target'; ignoring 'published: 8083'
 ! port 84: UDP ports default to 'host' mode (add 'mode: host' to silence)
 ! port 85: 'published' should be equal to 'target'; ignoring 'published: 8085'
 ! port 85: UDP ports default to 'host' mode (add 'mode: host' to silence)
 ! service "long": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "long": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "long-published": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "long-published": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "short": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "short": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "short-published": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "short-published": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "short-udp": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "short-udp": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "short-udp-published": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "short-udp-published": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors