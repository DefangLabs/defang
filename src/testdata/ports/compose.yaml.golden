name: ports
services:
  long:
    image: bogus
    networks:
      default: null
    ports:
      - mode: ingress
        target: 82
        protocol: tcp
  long-published:
    image: bogus
    networks:
      default: null
    ports:
      - mode: ingress
        target: 83
        published: "8083"
        protocol: tcp
  short:
    image: bogus
    networks:
      default: null
    ports:
      - mode: ingress
        target: 80
        protocol: tcp
  short-published:
    image: bogus
    networks:
      default: null
    ports:
      - mode: ingress
        target: 81
        published: "8081"
        protocol: tcp
  short-udp:
    image: bogus
    networks:
      default: null
    ports:
      - mode: ingress
        target: 84
        protocol: udp
  short-udp-published:
    image: bogus
    networks:
      default: null
    ports:
      - mode: ingress
        target: 85
        published: "8085"
        protocol: udp
networks:
  default:
    name: ports_default
